staging/src/k8s.io/apiserver/pkg/server/egressselector/egress_selector.go

```diff
@@ -239,17 +239,20 @@ func (d *dialerCreator) createDialer() utilnet.DialFunc {
 	if d.direct {
 		return directDialer
 	}
-	return func(ctx context.Context, network, addr string) (net.Conn, error) {
-		ctx, span := tracing.Start(ctx, fmt.Sprintf("Proxy via %s protocol over %s", d.options.protocol, d.options.transport), attribute.String("address", addr))
+	return func(requestCtx context.Context, network, addr string) (net.Conn, error) {
+		requestCtx, span := tracing.Start(requestCtx, fmt.Sprintf("Proxy via %s protocol over %s", d.options.protocol, d.options.transport), attribute.String("address", addr))
 		defer span.End(500 * time.Millisecond)
+		// Prevent goroutines from accumulating on connect(), since requestCtx may never cancel.
+		connectCtx, cancel := context.WithTimeout(requestCtx, time.Duration(10*time.Second))
```

Comment:
nit: don't need to cast to `time.Duration` since `time.Second` is already that type, and `10` is untyped.
```suggestion
		connectCtx, cancel := context.WithTimeout(requestCtx, 10*time.Second)
```

Reply:
Done